ipn/ipnlocal/local.go

```diff
@@ -606,6 +607,53 @@ func (b *LocalBackend) GetDNSOSConfig() (dns.OSConfig, error) {
 	return manager.GetBaseConfig()
 }
 
+// QueryDNS performs a DNS query for name and queryType using the built-in DNS resolver, and returns
+// the raw DNS response and the resolvers that are were able to handle the query (the internal forwarder
+// may race multiple resolvers).
+func (b *LocalBackend) QueryDNS(name string, queryType dnsmessage.Type) (res []byte, resolvers []*dnstype.Resolver, err error) {
+	manager, ok := b.sys.DNSManager.GetOK()
+	if !ok {
+		return nil, nil, errors.New("DNS manager not available")
+	}
+	if !strings.HasSuffix(name, ".") {
+		name += "."
+	}
```

Comment:
Any reason not to use `dnsname.ToFQDN`, as you do below, for the query? It has a `WithTrailingDot` method which will do this part, too 😃

Reply:
done